src/main/java/com/cloudant/search3/SearchServer.java

```diff
@@ -0,0 +1,273 @@
+package com.cloudant.search3;
+
+import static io.netty.buffer.Unpooled.wrappedBuffer;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONNECTION;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE;
+import static io.netty.handler.codec.http.HttpHeaderValues.APPLICATION_OCTET_STREAM;
+import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpMethod.POST;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+
+import java.io.IOException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.queryparser.classic.ParseException;
+
+import com.cloudant.search3.grpc.Search3.AnalyzeRequest;
+import com.cloudant.search3.grpc.Search3.DocumentDeleteRequest;
+import com.cloudant.search3.grpc.Search3.DocumentUpdateRequest;
+import com.cloudant.search3.grpc.Search3.ErrorResponse;
+import com.cloudant.search3.grpc.Search3.ErrorResponse.Type;
+import com.cloudant.search3.grpc.Search3.GroupSearchRequest;
+import com.cloudant.search3.grpc.Search3.Index;
+import com.cloudant.search3.grpc.Search3.SearchRequest;
+import com.cloudant.search3.grpc.Search3.SetUpdateSeqRequest;
```

Comment:
are we leaving the `grpc` package name in here because the generated files still somehow use that path?

Reply:
I kept the package name to reduce the size of the diff, I'm not averse to changing it.